integration-cli/docker_cli_build_test.go

```diff
@@ -7218,3 +7218,39 @@ func (s *DockerSuite) TestBuildLabelsOverride(c *check.C) {
 	}
 
 }
+
+// Shell test to confirm config gets updated correctly
+func (s *DockerSuite) TestBuildShellUpdatesConfig(c *check.C) {
+	name := "testbuildshellupdatesconfig"
+
+	expected := `["foo","-bar","#(nop) ","SHELL foo -bar"]`
+	_, err := buildImage(name,
+		`FROM `+minimalBaseImage()+`
+        SHELL foo -bar`,
+		true)
+	if err != nil {
+		c.Fatal(err)
+	}
+	res := inspectFieldJSON(c, name, "ContainerConfig.Cmd")
```

Comment:
Let's also inspect the `Shell` property to make sure the value is saved in the resulting image.